packages/flutter_tools/lib/src/asset.dart

```diff
@@ -389,11 +399,284 @@ class ManifestAssetBundle implements AssetBundle {
         )) font.descriptor,
     ];
   }
+
+  DevFSStringContent _createAssetManifest(Map<_Asset, List<_Asset>> assetVariants) {
```

Comment:
This change mostly takes the top-level private methods and makes them methods on the ManifestAssetBundle instead.

The other change is moving file creation out of the _Asset class to avoid the direct dep on the filesystem

Reply:
this code could be simplified further but I chose to keep it mostly as is